modyn/supervisor/internal/triggers/drift/alibi_detector.py

```diff
@@ -0,0 +1,107 @@
+from __future__ import annotations
+
+from typing import Union
+
+import alibi_detect.utils.pytorch
+import numpy as np
+import pandas as pd
+import torch
+from alibi_detect.cd import ChiSquareDrift, CVMDrift, FETDrift, KSDrift, LSDDDrift, MMDDrift
+from modyn.config.schema.pipeline import AlibiDetectDriftMetric, AlibiDetectMmdDriftMetric, MetricResult
+from modyn.config.schema.pipeline.trigger.drift.alibi_detect import AlibiDetectCVMDriftMetric, AlibiDetectKSDriftMetric
+from typing_extensions import override
+
+from .drift_detector import DriftDetector
+
+_AlibiMetrics = Union[
+    MMDDrift,
+    ChiSquareDrift,
+    KSDrift,
+    CVMDrift,
+    FETDrift,
+    LSDDDrift,
+]
+
+
+class AlibiDriftDetector(DriftDetector):
+    def __init__(self, metrics_config: dict[str, AlibiDetectDriftMetric]):
+        super().__init__(metrics_config)
+
+    @override
+    def init_detector(self) -> None:
+        pass
+
+    # to do: reuse reference data, and the configured metrics (incl. kernels)
+
+    @override
+    def detect_drift(
+        self,
+        embeddings_ref: pd.DataFrame | np.ndarray | torch.Tensor,
+        embeddings_cur: pd.DataFrame | np.ndarray | torch.Tensor,
```

Comment:
just for my understanding because we had this discussion before: do you want to run the detection on embeddings or the data directly or support both? (you mentioned alibi detect supports e.g. image input data)

Reply:
Currently, I'm only looking at embeddings in modyn. Raw data would probably need more architectural changes.

Reply:
I see. Let's see whether we need it, I hope it will not be too big architectural changes (should be mostly changing the flow within the alibi detector since the data should be there anyways), depends on the quality of the results. As soon as the selection stuff is written, I can also dive into coding here and join forces.